core/execution/market.go

```diff
@@ -754,28 +778,11 @@ func (m *Market) OnTick(ctx context.Context, t time.Time) bool {
 		if len(closedWithoutLP) > 0 {
 			m.releaseExcessMargin(ctx, closedWithoutLP...)
 		}
+		// last traded price should not reflect the closeout trades
+		m.lastTradedPrice = mp.Clone()
 	}
-
-	// check auction, if any. If we leave auction, MTM is performed in this call
-	m.checkAuction(ctx, t)
-	timer.EngineTimeCounterAdd()
-
-	m.updateMarketValueProxy()
-	m.updateLiquidityFee(ctx)
-	if m.settlement.HasTraded() && m.closed && mp != nil {
-		// we have trades, and the market has been closed. Perform MTM sequence now so the final settlement
-		// works as expected.
-		m.markPrice = mp.Clone()
-		mcmp := num.UintZero().Div(mp, m.priceFactor) // create the market representation of the price
-		dummy := &types.Order{
-			ID:            m.idgen.NextID(),
-			Price:         mp.Clone(),
-			OriginalPrice: mcmp,
-		}
-		m.confirmMTM(ctx, dummy, nil)
-	}
-	m.broker.Send(events.NewMarketTick(ctx, m.mkt.ID, t))
-	return m.closed
+	// update margin requirements
+	_ = m.recheckMargin(ctx, m.position.Positions())
```

Comment:
should we also releaseExcessMargin here?

Reply:
Wouldn't recheckMargin do that?

Reply:
In case we have trades, excess margin is released earlier on already. This call only release ensures that parties have enough margin to cover their current open position. I can't really think of a situation where we would end up with excess margin that isn't released here.

Reply:
@jeremyletang - no
@EVODelavega - maybe if orders get cancelled in the book? idk.

Reply:
We had a call where we discussed this briefly. If a party cancels all orders from the book (and has no open position), their margin is released immediately. We did conclude, however, that with this periodic MTM change, there are cases where margin will not always get released instantly, and that this was acceptable.